coding_systems/base/import_data_utils.py

```diff
@@ -136,3 +149,118 @@ def setup_new_import_database(self):
         call_command("migrate", self.coding_system, database=database_alias)
 
         return database_alias
+
+
+def update_codelist_version_compatibility(coding_system_id, new_database_alias):
+    new_coding_system = CODING_SYSTEMS[coding_system_id](
+        database_alias=new_database_alias
+    )
+    """
+    Check compatibility of existing codelist versions with a new coding system release
+
+    Filter codelist versions to only those that are compatible with, or created from,
+    the previous release. Any codelist versions that were not compatible with the previous
+    release can ignored, as we wouldn't expect them to be compatible with any later releases.
+
+    Note we don't exlude version created from or already in the compatible
+    set for THIS release.  If we're doing a forced-overwrite import (which is where
+    we'd expect that might happen), we want to re-check compatibility
+
+    Then check
+    1) versions with searches by rerunning the searches with the new release and checking
+       the results are the same
+    2) versions without searches, but with hierarchies, by recreating the hierarchy with the
+       new release and checking if it is identical to the version's cached hierarchy
+    """
+    previous_release = (
+        CodingSystemRelease.objects.filter(coding_system=coding_system_id)
+        .exclude(id=new_coding_system.release.id)
+        .latest("valid_from")
+    )
+    # Filter to versions compatible with previous release
+    versions_to_check = CodelistVersion.objects.filter(
+        Q(compatible_releases__id=previous_release.id)
+        | Q(coding_system_release=previous_release)
+    )
+    print(
+        f"Found {len(versions_to_check)} versions compatible with previous release '{previous_release.database_alias}'."
+    )
+    # Find any versions with searches
+    print("Checking compatibility by search:")
+    versions_with_searches = set(
+        versions_to_check.filter(
+            codelist__coding_system_id=coding_system_id, searches__isnull=False
+        )
+    )
```

Comment:
I don't think this is enough: you can have a codelist version that was created by uploading a CSV, and where a user has done some searches, but where these searches don't cover all of the codes in the codelist.

So should we be only checking the hierarchies?

Reply:
So should we only check by searches for codelist versions where the searches cover all codes?